specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2023-12-10-preview/cloudhsm.json

```diff
@@ -0,0 +1,967 @@
+            "name"
+          ],
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnection"
+          }
+        },
+        "statusMessage": {
+          "type": "string",
+          "description": "Cloud HSM Cluster status message",
+          "readOnly": true
+        },
+        "cloudHsmClusterRestoreProperties": {
+          "$ref": "#/definitions/CloudHsmClusterRestoreProperties"
+        }
+      },
+      "description": "Properties of a Cloud HSM Cluster.",
+      "type": "object"
+    },
+    "CloudHsmClusterPatchParameters": {
+      "properties": {
+        "tags": {
+          "type": "object",
+          "description": "The Cloud HSM Cluster's tags",
+          "additionalProperties": {
+            "type": "string",
+            "description": "The Cloud HSM Cluster's tag's additional properties"
+          }
+        },
+        "sku": {
+          "$ref": "#/definitions/CloudHsmClusterSku",
+          "description": "SKU details"
+        },
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v5/managedidentity.json#/definitions/ManagedServiceIdentity"
+        }
+      },
+      "description": "Patchable properties of the Cloud HSM Cluster",
+      "type": "object"
+    },
+    "CloudHsmClusterSecurityDomainProperties": {
+      "type": "object",
+      "description": "Security domain properties information for Cloud HSM cluster",
+      "properties": {
+        "fipsState": {
+          "format": "int32",
+          "type": "integer",
+          "description": "FIPS state information for security domain"
+        },
+        "activationStatus": {
+          "type": "string",
+          "description": "status of security domain activation"
+        }
+      }
+    },
+    "CloudHsmClusterSku": {
+      "type": "object",
+      "description": "Cloud Hsm Cluster SKU information",
+      "required": [
+        "name",
+        "family"
+      ],
+      "properties": {
+        "family": {
+          "type": "string",
+          "description": "Sku family of the Cloud HSM Cluster",
+          "enum": [
+            "B"
+          ],
+          "x-ms-client-default": "B",
+          "x-ms-enum": {
+            "name": "CloudHsmClusterSkuFamily",
+            "modelAsString": true
+          }
+        },
+        "name": {
+          "type": "string",
+          "description": "Sku name of the Cloud HSM Cluster",
+          "enum": [
+            "Standard_B1",
+            "Standard B10"
+          ],
+          "x-ms-enum": {
+            "name": "CloudHsmClusterSkuName",
+            "modelAsString": false
+          }
+        },
+        "capacity": {
+          "format": "int32",
+          "type": "integer",
+          "description": "Sku capacity"
+        }
+      }
+    },
+    "CloudHsmClusterBackupProperties": {
+      "type": "object",
+      "description": "Cloud Hsm Cluster backup information",
+      "properties": {
+        "storageResourceUri": {
```

Comment:
Let's close on this next week from Ritesh. I believe we need a container name also.

Reply:
I agree, we need container name unless we take the URI of the container itself. Should we give the storage account information its' own object similar to [Service Fabric's backup policy](https://github.com/Azure/azure-rest-api-specs/blob/b13bd252f5a0ae3e870dcd5fb4dc5c1389a7a734/specification/servicefabric/data-plane/Microsoft.ServiceFabric/stable/6.2/examples/CreateBackupPolicy-1.json#L18) so the storage configuration isn't mixed in with other backup-related properties?

Reply:
@sumantshiv, @JeffSimmer, I've made updates to the backup and restore properties schemas. After discussion with Sumant and Demi, we are following the same pattern with MHSM [data-plane APIs](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.4/backuprestore.json#L33C1-L49C13) 
Please let me know if everything looks good with the new changes